code/drasil-lang/lib/Language/Drasil/Constraint.hs

```diff
@@ -21,20 +21,34 @@ data Constraint a where
   -- | By default, physical and software constraints are ranges.
   Range          :: ConstraintReason -> RealInterval a a -> Constraint a
 
+  Elem           :: ConstraintReason -> [Expr] -> Constraint a
+
 -- | Smart constructor for range of 'Physical' constraints between two given expressions.
-physc :: RealInterval Expr Expr -> ConstraintE
-physc = Range Physical
+physRange :: RealInterval Expr Expr -> ConstraintE
+physRange = Range Physical
+
+physElem :: [Expr] -> ConstraintE
+physElem = Elem Physical
 
 -- | Smart constructor for range of 'Software' constraints between two given expressions.
-sfwrc :: RealInterval Expr Expr -> ConstraintE
-sfwrc = Range Software
+sfwrRange :: RealInterval Expr Expr -> ConstraintE
+sfwrRange = Range Software
+
+sfwrElem :: [Expr] -> ConstraintE
+sfwrElem = Elem Software
 
-isPhysC, isSfwrC :: Constraint e -> Bool
+isPhysRange, isSfwrRange, isPhysElem, isSfwrElem :: Constraint e -> Bool
 
 -- | Helpful for filtering for Physical constraints. True if constraint is 'Physical'.
-isPhysC (Range Physical _) = True
-isPhysC _ = False
+isPhysRange (Range Physical _) = True
+isPhysRange _ = False
+
+isPhysElem (Elem Physical _) = True
+isPhysElem _ = False
```

Comment:
Is the differentiation needed between Range and Elem for this? Why are the `isX` functions needed at all?

Reply:
It is used for formatting constraints. I don't think a distinction between the two would need to be made here
https://github.com/JacquesCarette/Drasil/blob/eacde82083e33e4c566cbac8b68373dfb90d35db/code/drasil-docLang/lib/Drasil/Sections/SpecificSystemDescription.hs#L278-L284
https://github.com/JacquesCarette/Drasil/blob/eacde82083e33e4c566cbac8b68373dfb90d35db/code/drasil-code/lib/Language/Drasil/Chunk/ConstraintMap.hs#L22-L28